### ISSUE TITLE
How to use aimet sdk to deploy quantized model in Qualcomm Neural Processing SDK

### ISSUE THREAD

**Issue by hammer422** [quic/aimet#1301]
I haven't found a complete example that can guide aimet how to deploy to the target device
So I did a simple test, but I found that the result of aimet was inconsistent with that of DSP

`
class Net(nn.Module):

    def __init__(self):
        super(Net, self).__init__()
        self.avg_pool = nn.AdaptiveAvgPool2d(1)
        self.fc1 = nn.Linear(3, 512)
        self.fc2 = nn.Linear(512, 10)
    def forward(self, x): # (1,3,20,20)
        x = self.avg_pool(x)
        x = x.view(x.size(0), -1)
        x = self.fc1(x)
        x = self.fc2(x)
        return x
`

**After using CLE and bias_correction，i just export QuantizationSimModel to onnx with encoding file**
`    
def export_quantsim(model:nn.Module, evalutor):

        dummy_input = torch.rand((1,3,20,20))
        iterations = RAW_DATA_LIST_LEN // BATCH_SIZE

        quantsim = QuantizationSimModel(model=model, quant_scheme='tf_enhanced',
                                        dummy_input=dummy_input, rounding_mode='nearest',
                                        default_output_bw=8, default_param_bw=8, in_place=False)

        quantsim.compute_encodings(forward_pass_callback=partial(evalutor, show=True),
                                   forward_pass_callback_args=iterations)

        from aimet_torch.quantsim import OnnxExportApiArgs

        onnx_params = OnnxExportApiArgs(opset_version=None, input_names=["inp"], output_names=["logits"])
        quantsim.export("./aimet_quat/tmp/", "tmp", dummy_input, onnx_params)`

**and now, i invoke snpe dlc converter** 

`./snpe-onnx-to-dlc --input_network tmp.onnx --output_path .tmp.dlc --quantization_overrides tmp.encodings`

`./snpe-dlc-quantize --input_dlc tmp.dlc --input_list tmp_file.txt --output_dlc tmp_quat.dlc --override_params`

**Finally, i push all relevant file to my test phone and run**
`./snpe-net-run --container /sdcard/tmp_quat.dlc --input_list /sdcard/tmp_file.txt --use_dsp`

**Here are my results**
input_data =  np.ones((1,3,20,20), dtype=np.float32) * 0.5

pytorch original model : 
logits = model(input_data) # tensor([-0.0676,  0.1706,  0.1283, -0.2093,  0.3666, -0.1050, -0.6430, -0.1969,0.6287,  0.0333])

QuantizationSimModel:
logits = quatsim_model(input_data) # tensor[-0.06763621  0.1706112   0.12825228 -0.2092978   0.36656314 -0.10497208 -0.6429729  -0.1969333   0.6287056   0.03326665]

snpe-net-run on dsp:
adb pull tmp_output/ .
logits = np.fromfile("tmp_output/Result_0/logits.raw", dtype=np.float32)
logits
[-0.10431736, 0.23324722, -0.01986997, 0.08444739, 0.49674934, 0.00496749, -0.05464243, 0.0745124, 0.26327714, -0.12418734]

The result of DSP is completely wrong. Why and whether I did something wrong

**Comment by gd2016229035** [comment]
@changewOw Hi, How did you solve this problem?

**Comment by pubuzai** [comment]
> I haven't found a complete example that can guide aimet how to deploy to the target device So I did a simple test, but I found that the result of aimet was inconsistent with that of DSP
> 
> ` class Net(nn.Module):
> 
> ```
> def __init__(self):
>     super(Net, self).__init__()
>     self.avg_pool = nn.AdaptiveAvgPool2d(1)
>     self.fc1 = nn.Linear(3, 512)
>     self.fc2 = nn.Linear(512, 10)
> def forward(self, x): # (1,3,20,20)
>     x = self.avg_pool(x)
>     x = x.view(x.size(0), -1)
>     x = self.fc1(x)
>     x = self.fc2(x)
>     return x
> ```
> 
> `
> 
> **After using CLE and bias_correction，i just export QuantizationSimModel to onnx with encoding file** ` def export_quantsim(model:nn.Module, evalutor):
> 
> ```
>     dummy_input = torch.rand((1,3,20,20))
>     iterations = RAW_DATA_LIST_LEN // BATCH_SIZE
> 
>     quantsim = QuantizationSimModel(model=model, quant_scheme='tf_enhanced',
>                                     dummy_input=dummy_input, rounding_mode='nearest',
>                                     default_output_bw=8, default_param_bw=8, in_place=False)
> 
>     quantsim.compute_encodings(forward_pass_callback=partial(evalutor, show=True),
>                                forward_pass_callback_args=iterations)
> 
>     from aimet_torch.quantsim import OnnxExportApiArgs
> 
>     onnx_params = OnnxExportApiArgs(opset_version=None, input_names=["inp"], output_names=["logits"])
>     quantsim.export("./aimet_quat/tmp/", "tmp", dummy_input, onnx_params)`
> ```
> 
> **and now, i invoke snpe dlc converter**
> 
> `./snpe-onnx-to-dlc --input_network tmp.onnx --output_path .tmp.dlc --quantization_overrides tmp.encodings`
> 
> `./snpe-dlc-quantize --input_dlc tmp.dlc --input_list tmp_file.txt --output_dlc tmp_quat.dlc --override_params`
> 
> **Finally, i push all relevant file to my test phone and run** `./snpe-net-run --container /sdcard/tmp_quat.dlc --input_list /sdcard/tmp_file.txt --use_dsp`
> 
> **Here are my results** input_data = np.ones((1,3,20,20), dtype=np.float32) * 0.5
> 
> pytorch original model : logits = model(input_data) # tensor([-0.0676, 0.1706, 0.1283, -0.2093, 0.3666, -0.1050, -0.6430, -0.1969,0.6287, 0.0333])
> 
> QuantizationSimModel: logits = quatsim_model(input_data) # tensor[-0.06763621 0.1706112 0.12825228 -0.2092978 0.36656314 -0.10497208 -0.6429729 -0.1969333 0.6287056 0.03326665]
> 
> snpe-net-run on dsp: adb pull tmp_output/ . logits = np.fromfile("tmp_output/Result_0/logits.raw", dtype=np.float32) logits [-0.10431736, 0.23324722, -0.01986997, 0.08444739, 0.49674934, 0.00496749, -0.05464243, 0.0745124, 0.26327714, -0.12418734]
> 
> The result of DSP is completely wrong. Why and whether I did something wrong

@changewOw Hi~, I encountered the same issue. Have you resolved it?
